src/viam/resource/registry.py

```diff
@@ -111,25 +128,7 @@ def register_resource_creator(cls, subtype: "Subtype", model: "Model", creator:
         with cls._lock:
             if key in cls._RESOURCES:
                 raise DuplicateResourceError(key)
-            cls._RESOURCES[key] = creator
-
-    @classmethod
-    def register_resource_validator(cls, subtype: "Subtype", model: "Model", validator: "Validator"):
-        """Register a specific validator function
-
-        Args:
-            subtype (Subtype): The Subtype of the resource
-            model (Model): The Model of the resource
-            validator (Validator): A function that can validate a resource and return implicit dependencies
-
-        Raises:
-            DuplicateResourceError: Raised if the Subtype and Model pairing already has a registered validator
-        """
-        key = f"{subtype}/{model}"
-        with cls._lock:
-            if key in cls._VALIDATORS:
-                raise DuplicateResourceError(key)
-            cls._VALIDATORS[key] = validator
+            cls._RESOURCES[key] = registration
```

Comment:
we will want to check that resource creator is not empty in this case - validator can be empty tho

Reply:
we can keep as is for now - we're placing more burden on devs but if they have some kind of linting turned on it should be fine